suite-native/module-settings/src/components/SettingsSectionItem.tsx

```diff
@@ -1,11 +1,53 @@
-import { ListItem, ListItemProps } from '@suite-native/atoms';
-
-export const SettingsSectionItem = ({ title, subtitle, iconName, onPress }: ListItemProps) => (
-    <ListItem
-        onPress={onPress}
-        subtitle={subtitle}
-        title={title}
-        iconName={iconName}
-        hasRightArrow
-    />
-);
+import { ReactNode } from 'react';
+import { ActivityIndicator, TouchableOpacity, View } from 'react-native';
+
+import { Box } from '@suite-native/atoms';
+import { Icon, IconName } from '@suite-common/icons-deprecated';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { SettingsSectionItemIcon } from './SettingsSectionItemIcon';
+import { SettingsSectionItemText } from './SettingsSectionItemText';
+
+export type SettingsSectionItemProps = {
+    iconName?: IconName;
+    title: ReactNode;
+    subtitle?: ReactNode;
+    onPress?: () => void;
+    isLoading?: boolean;
+    testID?: string;
+};
+
+const listItemRightContainerStyle = prepareNativeStyle(utils => ({
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginLeft: utils.spacings.medium,
+}));
+
+export const SettingsSectionItem = ({
+    title,
+    subtitle,
+    iconName,
+    onPress,
+    testID,
+    isLoading,
+}: SettingsSectionItemProps) => {
+    const { applyStyle, utils } = useNativeStyles();
+
+    const isDisabled = !onPress || isLoading;
```

Comment:
no need to have here onPress, since default value for this is `() => void` making it always false